### PR TITLE
dnd-tools: init at unstable-2021-02-18

### LINKED ISSUE #128076: Add `dnd-tools` package
**Project description**
DnD tools is a set of interactive CLI tools for DnD, a RPG.

**Metadata**
* source URL: [savagezen/dnd-tools](https://github.com/savagezen/dnd-tools)
* license: AGPL-3.0
* platforms: cross-platform
* dependencies: python (runtime)

### PR DESCRIPTION
**Pull request by urlordjames**
###### Motivation for this change
resolve #128076 because it's a pretty trivial package

#### things I may have done wrong:
- Upstream's `setup.py` says the version is `0.1.5`, but this has not been updated since 2018, so I use short commit sha instead.  I'm not sure if this is the proper way to do this.
- I'm not entirely sure what the difference between `licenses.agpl3Only` and `licenses.agpl3Plus` is, so the license is potentially inaccurate.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129496` at 0acd8032 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>dnd-tools</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/applications/misc/dnd-tools/default.nix:24:0:
```
   |
24 |     description = "A set of interactive command line tools for Dungeons and Dragons 5th Edition";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/dnd-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ python3, fetchFromGitHub, fetchpatch, lib }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "dnd-tools";
+  version = "baefb9e";
+
+  src = fetchFromGitHub {
+    owner = "savagezen";
+    repo = "dnd-tools";
```
Comment (musfay) [inline pkgs/applications/misc/dnd-tools/default.nix:9]:
```suggestion
    repo = pname;
```

File: pkgs/applications/misc/dnd-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ python3, fetchFromGitHub, fetchpatch, lib }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "dnd-tools";
+  version = "baefb9e";
```
Comment (musfay) [inline pkgs/applications/misc/dnd-tools/default.nix:5]:
```suggestion
  version = "unstable-2021-02-19";
```

**Review by musfay — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/dnd-tools/default.nix (hunk 2 above)
Comment (urlordjames) [reply pkgs/applications/misc/dnd-tools/default.nix:5]: I take it you meant `unstable-2021-02-18` right?

**Comment by urlordjames** [comment]
@musfay the license is correct right?  I just want to double check.

File: pkgs/applications/misc/dnd-tools/default.nix (hunk 2 above)
Comment (musfay) [reply pkgs/applications/misc/dnd-tools/default.nix:5]: 18 will be better. GitHub shows as 19 Feb but actual commit date is 18th.

Comment (urlordjames) [reply pkgs/applications/misc/dnd-tools/default.nix]: GitHub shows 18th for me, maybe a bug with timezones or something idk.

**Review by musfay — APPROVED** [review]

**Comment by musfay** [comment]
> @musfay the license is correct right? I just want to double check.

Yes. It's correct.

File: pkgs/applications/misc/dnd-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      sha256 = "00k8rsz2aj4sfag6l313kxbphcb5bjxb6z3aw66h26cpgm4kysp0";
+    })
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/savagezen/dnd-tools";
+    description = "A set of interactive command line tools for Dungeons and Dragons 5th Edition";
+    license = licenses.agpl3Only;
+    maintainers = [ maintainers.urlordjames ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/dnd-tools/default.nix:27]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**